File: java/src/main/java/problems1_100/RemoveElement.java
package problems1_100;

enum RemoveElement {
    DEFAULT {
        @Override
        int removeElement(int[] nums, int val) {
            int start = 0;
            int end = 0;
            while (start < nums.length) {
                if (nums[start] != val) {
                    nums[end] = nums[start];
                    end++;
                }
                start++;
            }
            return end;
        }
    };

    abstract int removeElement(int[] nums, int val);
}

File: kotlin/src/main/kotlin/problems1_100/RemoveElement.kt
package problems1_100

enum class RemoveElement {
    DEFAULT {
        override fun removeElement(nums: IntArray, `val`: Int): Int {
            var start = 0
            var end = 0
            while (start < nums.size) {
                if (nums[start] != `val`) {
                    nums[end] = nums[start]
                    end++
                }
                start++
            }
            return end
        }
    },
    ;

    abstract fun removeElement(nums: IntArray, `val`: Int): Int
}
